File: CuponRequest/src/main/java/com/example/CuponRequest/devtools/math/MathOperation.java
package com.example.CuponRequest.devtools.math;

import lombok.ToString;

public class MathOperation {
    // Вычисление процентного изменения
    public static Integer calculatePercentageChange(double oldValue, double newValue) {
        Integer percentageChange = Math.toIntExact(Math.round(Math.abs(((newValue - oldValue) / oldValue) * 100)));

        return percentageChange;
    }
}

File: CuponRequest/src/main/java/com/example/CuponRequest/devtools/math/MathOperation.kt
package com.example.CuponRequest.devtools.math

object MathOperation {
    // Вычисление процентного изменения
    fun calculatePercentageChange(oldValue: Double, newValue: Double): Int {
        return Math.toIntExact(Math.round(Math.abs((newValue - oldValue) / oldValue * 100)))
    }
}